### PR TITLE
update doc

### LINKED ISSUE #3791: Add partition env variable to Getting Started Guide
### Is an existing page relevant?

https://karpenter.sh/preview/getting-started/getting-started-with-karpenter/

### What karpenter features are relevant?

Using a partition other than `aws` in the getting started guide.

### How should the docs be improved?

The example in Step 3 of the getting started guide identifies `aws` as the partition type in the arn examples. Instead, in Step 2 an environment variable should be set (as is done in the [Migrating from Cluster Autoscaler](https://karpenter.sh/preview/getting-started/migrating-from-cas/) docs):

`AWS_PARTITION="aws" # if you are not using standard partitions, you may need to configure to aws-cn / aws-us-gov`

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by prabhjotsingh18**
docs:  website/content/en/preview/getting-started/getting-started-with-karpenter/

Fixes #3791

**Description**

updated getting-started-with-karpenter scripts changing hardcoded  to use variable AWS_PARTITION

**Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [X] Yes, issue opened: #3791
- [ ] No

**Release Note**
"NONE". 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/getting-started/getting-started-with-karpenter/scripts/step02-create-cluster.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ iam:
       namespace: karpenter
     roleName: ${CLUSTER_NAME}-karpenter
     attachPolicyARNs:
-    - arn:aws:iam::${AWS_ACCOUNT_ID}:policy/KarpenterControllerPolicy-${CLUSTER_NAME}
+    - arn:${AWS_PARTITION}:iam::${AWS_ACCOUNT_ID}:policy/KarpenterControllerPolicy-${CLUSTER_NAME}
```
Comment (njtran) [inline website/content/en/preview/getting-started/getting-started-with-karpenter/scripts/step02-create-cluster.sh:27]: Looks like there are two more occurrences on line 31 and line 53, can you add those in?

**Review by njtran — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks like there are some changes we can make in the migrating-from-cas guide too. Are you able to make those changes?

- 4 in `migrating-from-cas/scripts/step03-node-policies.sh`
- 1 in `migrating-from-cas/scripts/step05-controller-iam.sh`

**Review by jonathan-innis — COMMENTED** [review]
I think there's one more change that needs to be made to add the `${AWS_PARTITION}` variable [here](https://github.com/aws/karpenter/blob/main/website/content/en/preview/getting-started/migrating-from-cas/_index.md?plain=1#L83) but otherwise, LGTM 🚀

**Comment by jonathan-innis** [comment]
@prabhjotsingh18 Just checking that you saw my comment on the need to update the `AWS_PARTITION` in one more place

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by njtran — APPROVED** [review]

**Pull request merged**